cfgov/unprocessed/js/modules/ClearableInput.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+// Required polyfills for IE9.
+if ( !Modernizr.classlist ) { require( '../modules/polyfill/class-list' ); } // eslint-disable-line no-undef, global-require, no-inline-comments, max-len
+
+// Required modules.
+var atomicCheckers = require( '../modules/util/atomic-checkers' );
+
+/**
+ * ClearableInput
+ * @class
+ *
+ * @classdesc Initializes a new ClearableInput molecule.
+ *
+ * @param {HTMLNode} element
+ *   The DOM element within which to search for the molecule.
+ * @returns {Object} A ClearableInput instance.
+ */
+function ClearableInput( element ) {
+  var BASE_CLASS = 'input-contains-label';
+
+  var _dom =
+    atomicCheckers.validateDomElement( element, BASE_CLASS, 'ClearableInput' );
+  var _inputDom = _dom.querySelector( 'input' );
+  var _clearBtnDom = _dom.querySelector( '.' + BASE_CLASS + '_after__clear' );
+
+  /**
+   * @returns {Object} The ClearableInput instance.
+   */
+  function init() {
+    _clearBtnDom.addEventListener( 'mousedown', _clearClicked );
+    _inputDom.addEventListener( 'keyup', _inputTyped );
+
+    _setClearBtnState( _inputDom.value );
+
+    return this;
+  }
+
+  /**
+   * Event handler for when the clear input label was clicked.
+   * @param {MouseEvent} event The event object for the mousedown event.
+   */
+  function _clearClicked( event ) {
+    _inputDom.value = _setClearBtnState( '' );
+    _inputDom.focus();
+
+    // Prevent event bubbling up to the input, which would blur otherwise.
+    event.preventDefault();
+  }
+
+  /**
+   * Event handler for when the user typed in the input.
+   */
+  function _inputTyped() {
+    _setClearBtnState( _inputDom.value );
+  }
+
+  /**
+   * @param {string} value - The input value in the search box.
+   * @returns {string} The input value in the search box.
+   */
+  function _setClearBtnState( value ) {
+    if ( value === '' ) {
+      _hideClearBtn();
+    } else {
+      _showClearBtn();
+    }
+    return value;
+  }
+
+  /**
+   * Add a hidden class to the input search label.
+   * Used when there is no text input.
+   */
+  function _hideClearBtn() {
```

Comment:
It's so that the implementation for how the clear button is shown/hidden is hidden from where it is called. The clear button could be moved off screen, be set invisible, set to hidden, etc. and the method call signature of where it was used wouldn't change. Also, having `add` to hide the label seems less clear than saying "hideClearBtn." Probably overkill in this case though, I can update it.